auth/pat/pat.go

```diff
@@ -0,0 +1,20 @@
+package pat
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+type PATAuth struct {
```

Comment:
why do you need to re-implement auth, if there's a concurrent [PR](https://github.com/databricks/databricks-sql-go/pull/63) on integrating with Go SDK, that solves the auth problem in a unified way across different tools?